### PR TITLE
Additional properties in code

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/lite/request-response/components-api/src/main/scala/pl/touk/nussknacker/engine/requestresponse/api/openapi/RequestResponseOpenApiSettings.scala [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+import pl.touk.nussknacker.engine.api.definition.{JsonParameterEditor, MandatoryParameterValidator}
+
 object RequestResponseOpenApiSettings {
 
   val OutputSchemaProperty = "outputSchema"
   val InputSchemaProperty = "inputSchema"
   val OPEN_API_VERSION = "3.1.0"
 
+  val additionalPropertiesConfig = Map(
+    InputSchemaProperty -> AdditionalPropertyConfig(Some("{}"), Some(JsonParameterEditor), Some(List(MandatoryParameterValidator)), Some("Input schema")),
```
Comment (dswiecki) [inline engine/lite/request-response/components-api/src/main/scala/pl/touk/nussknacker/engine/requestresponse/api/openapi/RequestResponseOpenApiSettings.scala:13]: val emptySchema = "{}"

File: docs/MigrationGuide.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ To see the biggest differences please consult the [changelog](Changelog.md).
   `helper-utils` to `default-helpers`, separated `MathUtils` from
   `components-utils` to `math-utils`, removed dependencies from `helper-utils`
 * [#3420](https://github.com/TouK/nussknacker/pull/3420) `DeploymentManagerProvider.typeSpecificInitialData` takes deploymentConfig `Config` now
+* [#3493](https://github.com/TouK/nussknacker/pull/3493) Added method `DeploymentManagerProvider.additionalPropertiesConfig`, which allows to configure additional scenario properties programmatically.
```
[the remark this answers is not in the document]
Comment (mproch) [reply docs/MigrationGuide.md:20]: we change API so there is a mention... Line above follows the same pattern...

File: engine/lite/request-response/runtime/src/main/scala/pl/touk/nussknacker/engine/requestresponse/management/RequestResponseDeploymentManager.scala [hunk 3]
```diff
@@ -75,6 +76,8 @@ class RequestResponseDeploymentManagerProvider extends DeploymentManagerProvider
 
   override def typeSpecificInitialData(config: Config): TypeSpecificInitialData = TypeSpecificInitialData(RequestResponseMetaData(None))
 
+  override def additionalPropertiesConfig(config: Config): Map[String, AdditionalPropertyConfig] = Map.empty
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/lite/request-response/runtime/src/main/scala/pl/touk/nussknacker/engine/requestresponse/management/RequestResponseDeploymentManager.scala:79]: we use it in K8s

File: ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/DeploymentManagerProvider.scala [hunk 4]
```diff
@@ -21,6 +22,8 @@ trait DeploymentManagerProvider extends NamedServiceProvider {
 
   def typeSpecificInitialData(config: Config): TypeSpecificInitialData
 
+  def additionalPropertiesConfig(config: Config): Map[String, AdditionalPropertyConfig]
```
[the remark this answers is not in the document]
Comment (mproch) [reply ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/DeploymentManagerProvider.scala]: :+1: good point, changed

File: engine/lite/request-response/components-api/src/main/scala/pl/touk/nussknacker/engine/requestresponse/api/openapi/RequestResponseOpenApiSettings.scala (hunk 1 above)
Comment (mproch) [reply engine/lite/request-response/components-api/src/main/scala/pl/touk/nussknacker/engine/requestresponse/api/openapi/RequestResponseOpenApiSettings.scala]: :+1: good point, changed

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala [hunk 5]
```diff
@@ -59,23 +61,25 @@ class K8sDeploymentManagerProvider extends DeploymentManagerProvider {
 
   private val steamingInitialMetData = TypeSpecificInitialData(LiteStreamMetaData(Some(1)))
 
-  override def typeSpecificInitialData(config: Config): TypeSpecificInitialData = {
+  override def typeSpecificInitialData(config: Config): TypeSpecificInitialData = forMode(config)(_ => steamingInitialMetData, config => (scenarioName: ProcessName, _: String) => {
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala]: ok, changed

**Review by arkadius — APPROVED** [review]

**Review by dswiecki — APPROVED** [review]

**Pull request merged**